src/rez/package_test.py

```diff
@@ -718,3 +841,163 @@ def print_summary(self):
+        # True
+        >>> _is_conflict([PackageRequest("foo-2")], [PackageRequest("foo-1+")])
+        # False
+
+    Args:
+        requires (tuple[iter[PackageRequest]]):
+            A series of Rez packages requests to check for conflicts.
+
+    Returns:
+        bool: If conflict, return True, otherwise, return False.
+
+    """
+    requirement = RequirementList(itertools.chain(*requires))
+
+    return requirement.conflict
+
+
+def _choose_test_object(package, test_requires, extra_packages=frozenset()):
+    """Find the most appropriate Rez object for ``rez-test --interactive``.
+
+    If ``package`` has variants, find a valid variant, given ``test_requires``
+    and ``extra_packages``.
+
+    Args:
+        package (Package):
+            The installed Rez package to query variant / test data from.
+        test_requires (set[PackageRequest], optional)
+            Include these Rez package versions in the check, if provided. If
+            the Rez test has no defined ``"requires"``, this parameter is empty.
+        extra_packages (set[PackageRequest], optional):
+            Include these Rez package versions in the check, if provided.  This
+            is usually empty unless the user explicitly asks for them, e.g.
+            using ``rez-test foo --interactive bar --extra-packages thing-1``.
+
+    Raises:
+        PackageTestError:
+            If ``test_requires`` or ``extra_packages`` result in a Rez request conflict.
+
+    Returns:
+        Package or Variant:
+            Get a specific variant to return. If ``package`` has no variants
+            defined then this function returns ``package``, instead.
+
+    """
+    variants = package.variants
+
+    if not variants:
+        return package
+
+    candidates = []
+
+    for variant_index, package_requests in enumerate(variants):
+        if not _is_conflict(package_requests, test_requires, extra_packages):
+            candidates.append(package.get_variant(variant_index))
+
+    if not candidates:
+        if not extra_packages:
+            raise PackageTestError(
+                "No valid variant could match your given Rez test requires."
+            )
+
+        raise PackageTestError(
+            "No valid variant could match your given Rez test requires + extra packages."
+        )
+
+    return candidates[0]
+
+
+def _pre_test_env_commands(test_name, variant, executor):
+    """Run package.py:pre_test_commands() if present.
+
+    Args:
+        test_name (str):
+            The Rez test name which will be ran.
+        variant (Package or Variant):
+            A Rez object which may define ``pre_test_commands``.
+        executor (RexExecutor):
+            An interpreter to load commands during the ``rez-env``.
+
+    """
+    _pre_test_commands(test_name, variant, executor)
+
+    if hasattr(variant, "index"):
+        executor.setenv("REZ_TEST_VARIANT_INDEX", variant.index)
```

Comment:
Should this block be moved in `_pre_test_commands`? It probably wouldn't hurt to have this variable defined in the normal "rez-test` env. Which basically mean that interactive and non interactive could use the same function.

Also, maybe `REZ_CURRENT_TEST_VARIANT_INDEX`?

Reply:
We could put it in `_pre_test_commands` and then delete `_pre_test_env_commands` probably. I don't have a strong opinion one way or another.

However I will definitely change it to REZ_CURRENT_*, to match the other environment variable. Thank you for the suggestion!

As for whether to move or stay, I'll leave that to other contributors